acs-integration-tests/src/test/java/com/ge/predix/integration/test/PolicyEvaluationCachingIT.java

```diff
@@ -0,0 +1,519 @@
+    }
+
+    /**
+     * This test makes sure that cached policy evaluation results are properly invalidated when a policy has multiple
+     * resource attribute URI templates that match the request.
+     */
+    @Test
+    public void testPolicyWithMultAttrUriTemplatatesEvalCache() throws Exception {
+        BaseSubject subject = MARISSA_V1;
+        PolicyEvaluationRequestV1 policyEvaluationRequest = this.policyHelper.createEvalRequest("GET",
+                MARISSA_V1.getSubjectIdentifier(), "/v1/site/1/plant/asset/1", null);
+        String endpoint = this.acsUrl;
+
+        String testPolicyName = null;
+        try {
+            this.privilegeHelper.putSubject(this.acsAdminRestTemplate, subject, endpoint, this.acsZone1Headers,
+                    this.privilegeHelper.getDefaultAttribute(), this.privilegeHelper.getDefaultOrgAttribute());
+            String policyFile = "src/test/resources/policies/multiple-attribute-uri-templates.json";
+            testPolicyName = this.policyHelper.setTestPolicy(this.acsAdminRestTemplate, this.acsZone1Headers, endpoint,
+                    policyFile);
+
+            ResponseEntity<PolicyEvaluationResult> postForEntity = this.acsAdminRestTemplate.postForEntity(
+                    endpoint + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
+                    new HttpEntity<>(policyEvaluationRequest, this.acsZone1Headers), PolicyEvaluationResult.class);
+
+            Assert.assertEquals(postForEntity.getStatusCode(), HttpStatus.OK);
+            PolicyEvaluationResult responseBody = postForEntity.getBody();
+            Assert.assertEquals(responseBody.getEffect(), Effect.NOT_APPLICABLE);
+
+            BaseResource siteResource = new BaseResource("/site/1");
+            siteResource.setAttributes(new HashSet<Attribute>(
+                    Arrays.asList(new Attribute[] { this.privilegeHelper.getDefaultOrgAttribute() })));
+            this.privilegeHelper.putResource(this.acsAdminRestTemplate, siteResource, endpoint, this.acsZone1Headers,
+                    this.privilegeHelper.getDefaultOrgAttribute());
+
+            postForEntity = this.acsAdminRestTemplate.postForEntity(endpoint + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
+                    new HttpEntity<>(policyEvaluationRequest, this.acsZone1Headers), PolicyEvaluationResult.class);
+
+            Assert.assertEquals(postForEntity.getStatusCode(), HttpStatus.OK);
+            responseBody = postForEntity.getBody();
+            Assert.assertEquals(responseBody.getEffect(), Effect.PERMIT);
+        } finally {
+            if (testPolicyName != null) {
+                this.policyHelper.deletePolicySet(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        testPolicyName, this.acsZone1Headers);
+                this.privilegeHelper.deleteSubject(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        subject.getSubjectIdentifier(), this.acsZone1Headers);
+            }
+        }
+    }
+
```

Comment:
Should we also test for a single URI template?  Test that cached decision is invalidated if resource is updated that matches a single URI template...